src/application/delete_savedata.c

```diff
@@ -0,0 +1,382 @@
+
+    data->bgConfig = BgConfig_Alloc(data->heapId);
+
+    GraphicsModes modes = sDeleteSave_GraphicsModes;
+    SetBothScreensModesAndDisable(&modes);
+
+    BgTemplate template = sDeleteSave_BgTemplate;
+    InitBgFromTemplate(data->bgConfig, GF_BG_LYR_MAIN_0, &template, GX_BGMODE_0);
+    BgClearTilemapBufferAndCommit(data->bgConfig, GF_BG_LYR_MAIN_0);
+    LoadUserFrameGfx2(data->bgConfig, GF_BG_LYR_MAIN_0, 0x1E2, 2, 0, data->heapId);
+    LoadUserFrameGfx1(data->bgConfig, GF_BG_LYR_MAIN_0, 0x1D9, 3, 0, data->heapId);
+    LoadFontPal0(GF_PAL_LOCATION_MAIN_BG, GF_PAL_SLOT_1_OFFSET, data->heapId);
+    BG_ClearCharDataRange(GF_BG_LYR_MAIN_0, 32, 0, data->heapId);
+    BG_SetMaskColor(GF_BG_LYR_MAIN_0, RGB(1, 1, 27));
+    BG_SetMaskColor(GF_BG_LYR_SUB_0, RGB(1, 1, 27));
+}
+
+static void DeleteSavedataApp_FreeBgConfig(DeleteSavedataApp_Data *data) {
+    ToggleBgLayer(GF_BG_LYR_MAIN_0, GF_PLANE_TOGGLE_OFF);
+    ToggleBgLayer(GF_BG_LYR_MAIN_1, GF_PLANE_TOGGLE_OFF);
+    ToggleBgLayer(GF_BG_LYR_MAIN_2, GF_PLANE_TOGGLE_OFF);
+    ToggleBgLayer(GF_BG_LYR_MAIN_3, GF_PLANE_TOGGLE_OFF);
+    ToggleBgLayer(GF_BG_LYR_SUB_0, GF_PLANE_TOGGLE_OFF);
+    ToggleBgLayer(GF_BG_LYR_SUB_1, GF_PLANE_TOGGLE_OFF);
+    ToggleBgLayer(GF_BG_LYR_SUB_2, GF_PLANE_TOGGLE_OFF);
+    ToggleBgLayer(GF_BG_LYR_SUB_3, GF_PLANE_TOGGLE_OFF);
+
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_MAIN_0);
+    FreeToHeap(data->bgConfig);
+}
+
+static void DeleteSavedataApp_SetupTextAndWindow(DeleteSavedataApp_Data *data) {
+    data->msgData = NewMsgDataFromNarc(MSGDATA_LOAD_LAZY, NARC_msgdata_msg, NARC_msg_msg_0007_bin, data->heapId);
+    ResetAllTextPrinters();
+
+    data->printState = PRINTSTATE_PRINT_TEXT;
+
+    AddWindow(data->bgConfig, &data->window, &sDeleteSave_WindowTemplate1);
+    FillWindowPixelRect(&data->window, 0xF, 0, 0, 216, 32);
+}
+
+static void DeleteSavedataApp_FreeTextAndWindow(DeleteSavedataApp_Data *data) {
+    RemoveWindow(&data->window);
+    DestroyMsgData(data->msgData);
+}
+
+static BOOL DeleteSavedataApp_DoMainTask(DeleteSavedataApp_Data *data) {
+    BOOL ret = FALSE;
+
+    switch (data->mainState) {
+    case MAINSTATE_ASK_TO_DELETE:
+        if (DeleteSavedataApp_PrintMessage(data, msg_0007_00000, TRUE, 4) != TRUE) {
+            break;
+        }
+
+        data->yesNoMenu = CreateYesNoMenu(data->bgConfig, &sDeleteSave_WindowTemplate2, 0x1D9, 3, 1, data->heapId);
+        data->mainState = MAINSTATE_HANDLE_INPUT;
+        break;
+    case MAINSTATE_HANDLE_INPUT: {
+        u32 result = Handle2dMenuInput_DeleteOnFinish(data->yesNoMenu, data->heapId);
+        switch (result) {
+        case LIST_NO_MULTIPLE_SCROLL:
+            data->mainState = MAINSTATE_ASK_TO_CONFIRM;
+            break;
+        case LIST_CANCEL:
+            data->mainState = MAINSTATE_EXIT;
+            break;
+        }
+        break;
+    }
+    case MAINSTATE_ASK_TO_CONFIRM:
+        if (DeleteSavedataApp_PrintMessage(data, msg_0007_00001, TRUE, 4) != TRUE) {
```

Comment:
not necessary but would be nice: add comment above with contents of ``msg_0007_00001``